### DISCUSSION TITLE
UI issues for assets with a large number of partitions

### DISCUSSION THREAD

**Discussion by the4thamigo-uk** [dagster-io/dagster#19802] [Q&A] [answered]
I have an hourly partitioned job, which has accumulated 1000's of partitions. This results in (1) the asset lineage just displaying 999+  partitions, (2) the  asset partitions view difficult to see missed partitions and (3) the launch backfilll dialog difficult to select specific partition ranges. I was wondering how people deal with assets that have many partitions?

One idea I had was to find a way to expire old partitions (say older than 90 days, but not sure how to achieve this or what is the best practice. Does anyone have any input on this?

**Comment by yuhan** [comment, marked as answer]
Firstly, we recommend limiting the number of partitions for each asset to 25,000 or fewer, to avoid slower load times in the UI. ([docs](https://docs.dagster.io/concepts/partitions-schedules-sensors/partitioning-assets#recommended-partition-limits)). 

Setting up some data retention seems like a reasonable path forward. To manage the deletion of associated files older than a certain period, you could set up a schedule that runs daily to delete any associated files older than your desired retention period. Here's a relevant discussion: https://github.com/dagster-io/dagster/discussions/15381

**Reply by the4thamigo-uk** [reply]
Hi @yuhan, deleting the data is not really my issue. I want to limit the number of partitions defined in in my partition definition to reduce the number of partitions displayed in the UI. However, it seems that there is no way to configure a partition definition to automatically reset the start date, say to keep only partitions for the last 90 days, so the number of partitions just keeps growing.

Also it says to limit the number of partitions, but it does not say how this is achieved...?
